sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -494,6 +494,78 @@ export class StudentService extends RecordDataModelService<Student> {
     return searchQuery.getMany();
   }
 
+  /**
+   * Search students based on the search criteria.
+   * @param searchCriteria options to search by firstName,
+   * lastName, appNumber or sin.
+   * @returns list of students.
+   */
+  async searchStudentApplicationForInstitution(
+    institutionId: number,
+    searchCriteria: {
+      firstName?: string;
+      lastName?: string;
+      appNumber?: string;
+      sin?: string;
+    },
+  ): Promise<Student[]> {
+    const searchQuery = this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.birthDate",
+        "user.firstName",
+        "user.lastName",
+        "sinValidation.sin",
+      ])
+      .innerJoin(
+        Application,
+        "application",
+        "application.student.id = student.id",
+      )
+
+      .innerJoin("application.location", "pirLocation")
+      .innerJoin("pirLocation.institution", "pirInstitution")
+
+      .leftJoin("application.currentAssessment", "studentAssessment")
+      .leftJoin("studentAssessment.offering", "offering")
+      .leftJoin("offering.institutionLocation", "offeringLocation")
+      .leftJoin("offeringLocation.institution", "offeringInstitution")
+
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .where("user.isActive = true")
+      .andWhere(
+        "(offeringInstitution.id = :institutionId or pirInstitution.id = :institutionId)",
+        { institutionId },
+      )
+      .andWhere("application.applicationStatus != :overwrittenStatus", {
```

Comment:
Are we not ignoring the draft applications? as we are using the left join above for all associations created by submitted application.

Reply:
I understood the context from our conversation.